test/functional/provider/nodejs_spec.lua

```diff
@@ -0,0 +1,68 @@
+local helpers = require('test.functional.helpers')(after_each)
+local eq, clear = helpers.eq, helpers.clear
+local missing_provider = helpers.missing_provider
+local command = helpers.command
+local write_file = helpers.write_file
+local eval = helpers.eval
+local sleep = helpers.sleep
+local funcs = helpers.funcs
+
+do
+  clear()
+  if missing_provider('node') then
+    pending(
+      "Cannot find the neovim nodejs host. Try :checkhealth",
+      function() end)
+    return
+  end
+end
+
+before_each(function()
+  clear()
+end)
+
+describe('nodejs', function()
+  it('can inspect', function()
+    eq(1, funcs['provider#node#can_inspect']())
+  end)
+end)
+
+describe('nodejs host', function()
+  it('works', function()
+    local fname = 'hello.js'
+    write_file(fname, [[
+      const socket = process.env.NVIM_LISTEN_ADDRESS;
+      const neovim = require('neovim');
+      const nvim = neovim.attach({socket: socket});
+      nvim.command('let g:job_out = "hello"');
+      nvim.command('call jobstop(g:job_id)');
+    ]])
+    command('let g:job_id = jobstart(["node", "'..fname..'"])')
+    sleep(5000)
+    eq('hello', eval('g:job_out'))
+    os.remove(fname)
```

Comment:
should go in `after_each()`

Reply:
Perhaps teardown is better to guarantee deletion for all temporary test files.